### PR TITLE
Add support ideaIU and WebStorm

### PR DESCRIPTION
**Pull request by vzhukovskii**
This changes proposal adds an ability to build IntelliJ Idea Ultimate Edition and WebStorm.

List of the highlighted improvements:

- provide build argument to specify product version to build

  `--build-argument PRODUCT_VERSION=2020.2.3` – allows to specify version which has to be used during image build. Note, that supported version should be 2018.1 or later.

- provide build argument to specify product name to build

  `--build-argument PRODUCT_NAME=ideaIU` – allows to specify product that has to be used during image build. 

  Product name might be:

  - ideaIC
  - ideaIU
  - WebStorm

- add `license-configuration.sh` which perform provision activation code for offline usage to JetBrains product

- create example of Kubernetes Secret to provision activation code for offline usage

- developer documentation located in `/doc` folder

- move configuration files to `/config` folder

- move sh scripts to `/scripts` folder

- decouple supervisor configuration into dedicate pieces that includes in `supervisord.conf`

- added `.dockerignore`

- update default version from 2020.2.2 to 2020.2.3

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
Cool! Great docs content!
Now, it can be used as a base to update our scant [docs](https://github.com/eclipse/che-docs) on bringing a different IDE in [Che](https://www.eclipse.org/che/docs/che-7/end-user-guide/using-alternative-ides-in-che/) / [CRW](https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces/2.2/html/end-user_guide/customizing-developer-environments_crw#using-alternative-ides-in-codeready-workspaces_crw).

The updated doc should be in a form of a hands-on step by step guide on how to bring a completely different IDE into Che/CRW, on the example of IntelliJ IDEA.

File: devfiles/meta.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 publisher: fbenoit
```
Comment (benoitf) [inline devfiles/meta.yaml:2]: I think here we can change the publisher (but it's not related to this PR)

File: doc/Developer-Guide.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+- `workspace.yaml` – workspace configuration. Do not forged to provide correct url to the `meta.yaml` file:
+
+  ```
+  metadata:
+    name: che-ideaiu
+  components:
+    - type: cheEditor
+      reference: '<url for the meta.yaml goes here>'
+      alias: theia-editor
```
Comment (benoitf) [inline doc/Developer-Guide.md:212]: alias should be intellij no ?

File: doc/Developer-Guide.md [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+  displayName:  IntelliJ IDEA Ultimate Edition
+  title:  IntelliJ IDEA Ultimate Edition (in browser using noVNC) as editor for Eclipse Che
+  description:  IntelliJ IDEA Ultimate Edition running on the Web with noVNC
+  icon: https://resources.jetbrains.com/storage/products/intellij-idea/img/meta/intellij-idea_logo_300x300.png
+  category: Editor
+  repository: https://github.com/che-incubator/che-editor-intellij-community
+  firstPublicationDate: "2020-10-27"
+  spec:
+    endpoints:
+    # hack (use dirigible endpoint name as it is part of liveness probes)
```
Comment (benoitf) [inline doc/Developer-Guide.md:236]: now the hack is gone https://github.com/eclipse/che/blob/bedef8307f72c710c3344ae27910d39b64e07a9b/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties#L165

File: doc/Developer-Guide.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+           - exposedPort: 8080
+       memoryLimit: "2048M"
+  
+  ```
+
+
+
+As an example showed in the picture: 
+
+![create-workspace-from-devfile](https://raw.githubusercontent.com/che-incubator/che-editor-intellij-community/master/doc/images/create-workspace-from-devfile.jpg)
```
Comment (benoitf) [inline doc/Developer-Guide.md:171]: images should go in a separate branch to not make source code too big
https://medium.com/@minamimunakata/how-to-store-images-for-use-in-readme-md-on-github-9fb54256e951

File: scripts/license-configuration.sh [hunk 5]
```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
```
Comment (benoitf) [inline scripts/license-configuration.sh:1]: script is not passing on https://www.shellcheck.net/

Comment (vzhukovskii) [reply scripts/license-configuration.sh:1]: Fixed.

>No issues detected!

File: doc/Developer-Guide.md (hunk 4 above)
Comment (vzhukovskii) [reply doc/Developer-Guide.md]: Moved images to branch `media`. It was created with option `--orphan`, so it's history is clear.

File: doc/Developer-Guide.md (hunk 2 above)
Comment (vzhukovskii) [reply doc/Developer-Guide.md]: Updated with product identifier.

File: devfiles/meta.yaml (hunk 1 above)
Comment (vzhukovskii) [reply devfiles/meta.yaml]: Updated with eclipse.

**Review by benoitf — APPROVED** [review]

File: doc/Developer-Guide.md (hunk 3 above)
Comment (vzhukovskii) [reply doc/Developer-Guide.md]: Updated with `intellij`.

**Pull request merged**